src/osd/OSD.cc

```diff
@@ -3984,12 +3822,13 @@ void OSD::load_pgs()
 	  assert(0 == "Missing map in load_pgs");
 	}
       }
-      pg = _open_lock_pg(pgosdmap, pgid);
+      pg = _make_pg(pgosdmap, pgid);
     } else {
-      pg = _open_lock_pg(osdmap, pgid);
+      pg = _make_pg(osdmap, pgid);
     }
     // there can be no waiters here, so we don't call wake_pg_waiters
 
+    pg->lock();
```

Comment:
What's the advantage of killing ```_open_lock_pg()``` if we have to open-code the locking anyway? Dislike. I know there's only one caller, but it's nice defensive coding that doesn't cost anything to have a single wrapper...

Reply:
I was having a hard time keeping track of what all of the helpers and their variations meant; consolidating on a simple call and making the lock call explicit made the code clearer.  The only reason to still have _lookup_lock_pg() is that the pg->is_deleted() check is there (although tbh I think that may no longer be needed; I didn't get into it).

Reply:
I take it back; still needed (because the pg lock is ordered outside of the shard lock, and lookup goes the opposite direction)

Reply:
Anyway, taking or not taking the pg lock is fundamental enough that the caller needs to understand it (either by calling pg->lock() or by calling _lookup_lock_pg instead of _lookup_pg).  is_delete() aside, the fact that pg lock is no longer ordered inside the pg_map_lock means the helpers no longer serve a useful purpose.  (They were introduced and made sense back when you took pg lock inside of pg_map_lock)